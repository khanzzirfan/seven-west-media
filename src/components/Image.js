import React from 'react';
import styled from 'styled-components';

const margin = 9;
const size = 38;

const Image = styled.img`
 margin-top:${margin}px;
 margin-left:${margin}px;
 width: ${size}px;
 height: ${size}px;
`;

export default Image;