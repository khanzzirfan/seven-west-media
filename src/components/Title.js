import React from 'react';
import styled from 'styled-components';

const marginLeft = 10;
const fontSize = 20;

const Title = styled.h2`
    color: white;
    margin: 0 0 0 ${marginLeft}px;
    font-size: ${fontSize}px;
    font-family: Roboto, sans-serif;
    display: flex;
    justify-content: center;
    flex-direction: column;
    text-align: center;
`;

export default Title;