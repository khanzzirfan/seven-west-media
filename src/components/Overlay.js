import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from './Image';
import Title from './Title';

const overlayViewHeight = 51;
const OverlayView = styled.div`
    display:flex;
    align-self: flex-end;
    background: rgba(0, 0, 0, 0.6);
    width: 100%;
    height: ${overlayViewHeight}px;
`;

const Overlay = ({ iconAlt, iconImage, ImageElement, TitleElement, title }) => (
    <OverlayView>
        <ImageElement src={iconImage} alt={iconAlt} />
        <TitleElement>{title}</TitleElement>
    </OverlayView>
);

Overlay.defaultProps = {
    iconAlt: '',
    iconImage: null,
    ImageElement: Image,
    TitleElement: Title,
    title: '',
};

Overlay.propTypes = {
    iconAlt: PropTypes.string,
    iconImage: PropTypes.node,
    ImageElement: PropTypes.func,
    TitleElement: PropTypes.func,
    title: PropTypes.string,
};

export default Overlay;