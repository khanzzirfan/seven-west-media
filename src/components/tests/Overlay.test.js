import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import Overlay from '../Overlay';
import Image from '../Image';
import Title from '../Title';
import logoImage from '../../assets/logo.png';

describe('components -> Overlay component', () => {
    let instance;
    let testRenderer;

    beforeAll(() => {
        testRenderer = TestRenderer.create(<Overlay
            iconAlt={"media alt"}
            iconImage={logoImage}
            ImageElement={Image}
            TitleElement={Title}
            title={"Home and Away"}
        />);
        instance = testRenderer.root;
    });

    it('should renders correctly and match to snapshot', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("should pass props to image component", () => {
        const iconAlt = "media alt";
        const title = "Home and Away";
        expect(instance.findByType(Image).props.src).toBe(logoImage);
        expect(instance.findByType(Image).props.alt).toBe(iconAlt);
    })

    it('should pass props to Title component correctly', () => {
        const iconAlt = "media alt";
        const title = "Home and Away";
        expect(instance.findByType(Title).props.children).toBe(title);
    });

    it('should render different image component', () => {
        const iconAlt = "media alt2";
        const title = "Home and Away2";
        const customComponent = () => <div />;

        const testRendererComponent = TestRenderer.create(<Overlay
            iconAlt={"media alt2"}
            iconImage={logoImage}
            ImageElement={customComponent}
            TitleElement={Title}
            title={"Home and Away2"}
        />).root;

        expect(testRendererComponent.findByType(customComponent).props.src).toBe(logoImage);
        expect(testRendererComponent.findByType(customComponent).props.alt).toBe(iconAlt);
    });

    it('should render different title component', () => {
        const iconAlt = "media alt2";
        const title = "Home and Away2";
        const customComponent = () => <div />;

        const testRendererComponent = TestRenderer.create(<Overlay
            iconAlt={"media alt2"}
            iconImage={logoImage}
            ImageElement={Image}
            TitleElement={customComponent}
            title={"Home and Away2"}
        />).root;

        expect(testRendererComponent.findByType(customComponent).props.children).toBe(title);
    });
});