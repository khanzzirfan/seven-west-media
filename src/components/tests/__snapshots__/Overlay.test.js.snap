// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> Overlay component should renders correctly and match to snapshot 1`] = `
.c1 {
  margin-top: 9px;
  margin-left: 9px;
  width: 38px;
  height: 38px;
}

.c2 {
  color: white;
  margin: 0 0 0 10px;
  font-size: 20px;
  font-family: Roboto,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background: rgba(0,0,0,0.6);
  width: 100%;
  height: 51px;
}

<div
  className="c0"
>
  <img
    alt="media alt"
    className="c1"
    src="logo.png"
  />
  <h2
    className="c2"
  >
    Home and Away
  </h2>
</div>
`;
